app/flatpak-builtins-config.c

```diff
@@ -227,12 +247,32 @@ get_config_key (const char *arg, GError **error)
   return NULL;
 }
 
+static FlatpakDir *
+get_config_dir(FlatpakDir *dir, ConfigKey *key)
+{
+  switch (key->scope)
+    {
+      case CONFIG_SCOPE_INSTALLATION:
+        return dir;
+      case CONFIG_SCOPE_USER:
+        return flatpak_dir_get_user ();
```

Comment:
`flaptak_dir_get_user()` return value must be freed by the called (of `get_config_dir()`). However `dir` that is returned in some other case isn't.

Translation this will either crash or leak.